### PR TITLE
[chore] prepare release 0.60.0

### LINKED ISSUE #6069: Prepare core release v0.60.0

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:** 

Prepare 0.60.0 release

**Tests**: I opened open-telemetry/opentelemetry-collector-contrib/pull/14120 for double checking that core will work on contrib

**Link to tracking Issue:** Fixes #6069

### CODE REVIEW

**Comment by mx-psi** [comment]
Integration test failure is not a concern, I described what is happening with more detail on #6076. I don't think this should block merging this PR: we know why it fails and it will stop failing once we tag 0.60.0

**Comment by mx-psi** [comment]
Blocked until we decide if we merge #6064

File: cmd/otelcorecol/builder-config.yaml
Comment (bogdandrutu) [inline cmd/otelcorecol/builder-config.yaml:28]: @jpkrohling after you approved the change to have a default config for the builder, you need to clean this up since now we no longer needed it correct?

Comment (jpkrohling) [reply cmd/otelcorecol/builder-config.yaml:28]: Great point, created #6077 to track this.

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**
